lib/ansible/modules/cloud/vultr/vultr_server_facts.py

```diff
@@ -0,0 +1,177 @@
+description:
+  - Gather facts about servers available.
+version_added: "2.7"
+author: "Yanis Guenane (@Spredzy)"
+extends_documentation_fragment: vultr
+'''
+
+EXAMPLES = r'''
+- name: Gather Vultr servers facts
+  local_action:
+    module: vultr_server_facts
+
+- name: Print the gathered facts
+  debug:
+    var: ansible_facts.vultr_server_facts
+'''
+
+RETURN = r'''
+---
+vultr_api:
+  description: Response from Vultr API with a few additions/modification
+  returned: success
+  type: complex
+  contains:
+    api_account:
+      description: Account used in the ini file to select the key
+      returned: success
+      type: string
+      sample: default
+    api_timeout:
+      description: Timeout used for the API requests
+      returned: success
+      type: int
+      sample: 60
+    api_retries:
+      description: Amount of max retries for the API requests
+      returned: success
+      type: int
+      sample: 5
+    api_endpoint:
+      description: Endpoint used for the API requests
+      returned: success
+      type: string
+      sample: "https://api.vultr.com"
+vultr_server_facts:
+  description: Response from Vultr API
+  returned: success
+  type: complex
+  contains:
+    "vultr_server_facts": [
+      {
+        "APPID": 0,
+        "DCID": 7,
+        "FIREWALLGROUPID": 0,
+        "OSID": 167,
+        "VPSPLANID": 201,
+        "allowed_bandwidth_gb": 1000,
+        "auto_backup_enabled": false,
+        "cost_per_month": 5.00,
+        "current_bandwidth_gb": 0,
+        "date_created": "2018-07-19 08:23:03",
+        "default_password": "p4ssw0rd!",
+        "disk": "Virtual 25 GB",
+        "id": 17241096,
+        "internal_ip": "",
+        "kvm_url": "https://my.vultr.com/subs/vps/novnc/api.php?data=OFB...",
+        "name": "ansibletest",
+        "os": "CentOS 7 x64",
+        "pending_charges": 0.01,
+        "power_status": "running",
+        "ram": "1024 MB",
+        "region": "Amsterdam",
+        "server_state": "ok",
+        "status": "active",
+        "tag": "",
+        "v4_gateway": "105.178.158.1",
+        "v4_main_ip": "105.178.158.181",
+        "v4_netmask": "255.255.254.0",
+        "v6_main_ip": "",
+        "v6_network": "",
+        "v6_network_size": "",
+        "v6_networks": [],
+        "vcpu_count": 1
+      }
+    ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vultr import (
+    Vultr,
+    vultr_argument_spec,
+)
+
+
+class AnsibleVultrServerFacts(Vultr):
+
+    def __init__(self, module):
+        super(AnsibleVultrServerFacts, self).__init__(module, "vultr_server_facts")
+
+        self.returns = {
+            "APPID": dict(convert_to='int'),
+            "DCID": dict(convert_to='int'),
+            "FIREWALLGROUPID": dict(convert_to='int'),
+            "OSID": dict(convert_to='int'),
+            "SUBID": dict(key='id', convert_to='int'),
+            "VPSPLANID": dict(convert_to='int'),
```

Comment:
I would prefer to return the actual names, not the IDs for DCID, SUBID,, VPSPLANID. similar the way we do for vultr_server https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vultr/vultr_server.py#L357 What do you think?

Reply:
done